test/functional/page_objects/settings_page.ts

```diff
@@ -486,6 +486,55 @@ export function SettingsPageProvider({ getService, getPageObjects }: FtrProvider
       await this.clickSaveScriptedField();
     }
 
+    async addRuntimeField(name: string, type: string, script: string) {
+      await this.clickAddField();
+      await this.setFieldName(name);
+      await this.setFieldType(type);
+      if (script) {
+        await this.setFieldScript(script);
+      }
+      await this.clickSaveField();
+      await this.closeIndexPatternFieldEditor();
+    }
+
+    async closeIndexPatternFieldEditor() {
+      await retry.waitFor('field editor flyout to close', async () => {
+        return !(await testSubjects.exists('euiFlyoutCloseButton'));
+      });
+    }
+
+    async clickAddField() {
+      log.debug('click Add Field');
+      await testSubjects.click('addField');
+    }
+
+    async clickSaveField() {
+      log.debug('click Save');
+      await testSubjects.click('fieldSaveButton');
+    }
+
+    async setFieldName(name: string) {
+      log.debug('set field name = ' + name);
+      await testSubjects.setValue('nameField', name);
+    }
+
+    async setFieldType(type: string) {
+      log.debug('set type = ' + type);
+      await testSubjects.setValue('typeField', type);
+    }
+
+    async setFieldScript(script: string) {
+      log.debug('set script = ' + script);
+      await (
+        await (await testSubjects.find('formatRow')).findAllByCssSelector(
```

Comment:
This is not actually enabling the "Set value" toggle to add a script.

```suggestion
        await (await testSubjects.find('valueRow')).findAllByCssSelector(
```